### DISCUSSION TITLE
No Media Player entity picture when casting from a non-MA source?

### DISCUSSION THREAD

**Discussion by yorb** [music-assistant/support#3208] [Q&A] [answered]
In HA, it appears that the MA versions of Google Cast media players don't show album artwork if music is played directly to the speakers from another service (e.g., casting from Spotify directly to the speaker itself):
![image](https://github.com/user-attachments/assets/fbee3b3a-c477-4d11-9cb2-f849db160f43)

I'm still not totally clear how the HA integrations work, so I may be missing something. When I installed MA, it made duplicates of all my Google Cast media players. I'm not really sure why, but it does have the benefit of showing MA categories in the media browser that pops up when you click Browse Media, so I guess I should use these entities in my dashboards. The problem is they don't get the entity_picture.

Is this a bug? Is it Google Cast players only? Am I just using this totally wrong? :D

**Comment by OzGav** [comment]
The MA player entities are required for all of the advanced MA actions such as` play_announcement`.

Interacting with third party applications is not currently supported. If you want to do that then there is no point in using the MA players. You don't need to install the integration if you don't need to MA actions or need access to HA players within MA.

**Reply by yorb** [reply]
Understood about the MA player entities being required for MA actions; that makes sense. Thank you. My question is just about the MA player entities displaying album artwork when the speaker is playing content that was not started from MA. Is this possible?

Here's a screenshot of the two players (HA on top, MA on bottom) for one of my speakers:
<img width="1185" alt="image" src="https://github.com/user-attachments/assets/a24a19e2-57af-4786-b7ce-e4e3fb1b3108">

As you can see, both entities have the artist and title, but only the HA entity has the picture (album art). It's a very minor issue, but it got me thinking that maybe I'm not using it as intended. Maybe what I should do is use the HA player entity for control, and then add another card that somehow lets me browse MA for this player. It would just be convenient if I could use the MA player in my dashboards and have the album art show up too.

Anyway, thank you!

**Reply by OzGav** [reply]
Tidal can be added to MA so we would expect you would use MA exclusively.

**Reply by yorb** [reply]
That makes sense when playing from a device running HA, such as a wall panel. But what about all the times I'm just casting from my phone, or a guest casts from their phone? Surely you don't expect them to install HA and load the MA webview on their device just to play music. :)

Anyway, I think you've answered my questions. I'll follow up if I end up with a good workaround for my specific use case. Thank you!

**Reply by OzGav** [reply]
We are looking at adding Spotify Connect soon. Tidal is not currently on the radar.

> Surely you don't expect them to install HA 

Anyone can access your MA frontend from their mobile device, they don’t need the HA app. You can either give them a login to your HA instance or enable port 8095 from the web server settings

**Reply by marcelveldt** [reply, marked as answer]
Soon, we'll have some more improvements in making MA more aware of external sources active on a device. Adding the picture is then something we can look into.

**Reply by HarvsG** [reply]
Similar problem, streaming a BBC iPlayer programme to my Nest Home display showed the artwork on the cast media player and within the MusicAsistant UI, but not on the HA media player card.
![image](https://github.com/user-attachments/assets/980c394f-f36a-4886-b522-9ca388511f8a)
![image](https://github.com/user-attachments/assets/2e80cb3b-12d2-4a62-baf4-4dd15c2e58b5)

**Reply by HarvsG** [reply]
Relevant issue: https://github.com/home-assistant/core/issues/142926

**Comment by MacSergey** [comment]
I use this integration https://github.com/AlexxIT/YandexStation to stream music from Yandex Station to HomePod through MA. everything works fine expect MA player gets no title or cover

![image](https://github.com/user-attachments/assets/f524c9bc-f424-49f8-bdd3-0349c4f54a37)

**Reply by Den-88** [reply]
I have exactly the same problem. Did you somehow manage to solve it?

**Comment by acimer83** [comment]
I have the same problem, playing remote-share music through Music Assistant using Denon AVR speakers - no artwork or artist/track title on TV but MA instances show artwork and artist/track title

**Comment by HarvsG** [comment]
Similar problem, streaming a BBC iPlayer programme to my Nest Home display showed the artwork on the cast media player and within the MusicAsistant UI, but not on the HA media player card.
![image](https://github.com/user-attachments/assets/980c394f-f36a-4886-b522-9ca388511f8a)
![image](https://github.com/user-attachments/assets/2e80cb3b-12d2-4a62-baf4-4dd15c2e58b5)

**Comment by TinyShark** [comment]
+1 to this functionality. Would be great to see album art when music is being streamed from a different device, or by the device itself. (in my case using Sonos s2 players with Alexa voice control built in).

**Comment by catohagen** [comment]
+1 for this, MA is fantastic with its library and sources + DSP , but sometimes when casting from external sources it would be great to have media information show up.
